Add release notes for Metals 0.10.2 (#2729)
scalameta/metals: website/blog/2021-04-20-tungsten.md

authorImageURL: https://github.com/tgodzik.png
---

We're happy to announce the release of Metals v0.10.2, which mainly adds support
for Scala 3.0.0-RC3 as well a couple of smaller improvements.

<table>
<tbody>
  <tr>
    <td>Commits since last release</td>
    <td align="center">62</td>
  </tr>
  <tr>
    <td>Merged PRs</td>
    <td align="center">30</td>
  </tr>
    <tr>
    <td>Contributors</td>
    <td align="center">8</td>
  </tr>
  <tr>
    <td>Closed issues</td>
    <td align="center">8</td>
  </tr>
  <tr>
    <td>New features</td>
    <td align="center">4</td>
  </tr>
</tbody>
</table>

For full details: https://github.com/scalameta/metals/milestone/37?closed=1

Metals is a language server for Scala that works with VS Code, Vim, Emacs,
Sublime Text and Eclipse. Metals is developed at the
[Scala Center](https://scala.epfl.ch/) and [VirtusLab](https://virtuslab.com)
with the help from [Lunatech](https://lunatech.com) along with contributors from
the community.

## TL;DR

Check out [https://scalameta.org/metals/](https://scalameta.org/metals/), and
give Metals a try!

- [Scala 3] Support for Scala 3.0.0-RC3
- Add code action to add/remove string interpolation.

> - Offer organize imports when on an unused import.

Reviewer: ```suggestion
- Offer to organize imports when on an unused import.
```

Author: Fixed

- [Scala 3] Better names for types in completions and hovers.

## Add/remove string interpolation

Thanks to some great work by our newest contributor
[Giggiux](https://github.com/Giggiux) it's now possible to:

- easily add string interpolation to an existing string

```scala
val message = "Hello this is a message"
```

> when cursor is positioned inside the string we can invoke the code action and

Reviewer 2: ```suggestion
When the cursor is positioned inside of the string we can invoke the code action and
```

Author: Fxied

get:

```scala
val message = s"Hello this is a message"
```

> This is a simple action, but together with completions it makes sure that users
> don't waste too much time on it.

Reviewer 2: I'm not 100% sure what is meant here, so you may want to rephrase it a bit. After reading it I'm left wondering, "too much time on what?" Maybe something along the lines of:

```suggestion
While this is a simpler code action, it can be paired with other things like completions that allow for a workflow with less manual steps.
```

Author: Och, that sounds much better!

- remove string interpolation if no splices are found

> when a cursor is positioned inside a string interpolation, that does not contain

Reviewer 2: ```suggestion
When the cursor is positioned inside a string interpolation, that does not contain
```

Author: Fixed

any splices ( `$identifier`), we will now offer an action to quickly get rid of
the additional `s`.

```scala
val message = s"Hello this is a message without splices"
```

will change to:

```scala
val message = "Hello this is a message without splices"
```

## Offer organize imports when on an unused import

> Organize imports code action was already available for a while, however it's not

Reviewer 2: ```suggestion
The organize imports code action was already available for a while, however it's not
```

Author: Fixed

always easy to figure out how to run it. To make it easier for users to find
Metals now offers a code action to run organize imports whenever the cursor is

> situated on an actual unused import reported by the diagnostics. Thanks

Reviewer 3: I would somehow reformulate this because for me it sounds a little bit grumpy :smiley: .
Smth like: _we had to add a new action just because a lot of people still avoid figuring out how to run organize import_

Reviewer: Maybe something like:
> In addition to the usual "organize imports", to try to improve the user experience, now Metals offers a code action to organize import whenever the cursor is situated on an unused import reported by the diagnostics. 

without comments to "why" it was implemented?

Author: Fixed according to the suggestion by @Giggiux , thanks!

[ckipp01](https://github.com/ckipp01) for this new feature!

## [Scala 3] Better names for types in completions and hovers.

Thanks to another great PR by [tanishiking](https://github.com/tanishiking)
Metals will offer completions and hovers that include shorter names instead of
fully qualified ones whenever they are currently in scope. They will also be
more similar to the ones that you might be accustomed to in Scala 2. You can
find some more examples of improvements in the
[pull request](https://github.com/scalameta/metals/pull/2637) itself.

## Miscellaneous

- Make insert type work with renamed imports.

> - Treat `.srcjar` as jars to support Bazel in future (thanks

Reviewer 2: ```suggestion
- Treat `.srcjar` as jars to support Bazel in the future (thanks
```

Author: Fixed
